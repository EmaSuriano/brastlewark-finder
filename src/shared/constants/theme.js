export default {
  background: '#4F5D75',
  light: '#FFFFFF',
  disabled: '#BFC0C0',
  accent: '#EF8354',
  secondary: '#2D3142',
};
