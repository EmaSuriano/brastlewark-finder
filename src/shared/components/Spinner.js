import SpinnerSvg from 'react-svg-spinner';
import styled from 'styled-components';
import { ThemePropTypes } from 'shared/constants/theme';

const Spinner = styled(SpinnerSvg).attrs({
  size: '128px',
  color: props => props.theme.accent,
})``;

Spinner.propTypes = {
  theme: ThemePropTypes,
};

Spinner.displayName = 'Spinner';

export default Spinner;
