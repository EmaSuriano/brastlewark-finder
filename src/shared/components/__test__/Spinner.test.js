import React from 'react';
import { shallow } from 'enzyme';
import theme from 'shared/constants/theme';
import Spinner from '../Spinner';

describe('<Spinner />', () => {
  it('render', () => {
    expect(shallow(<Spinner theme={theme} />)).toMatchSnapshot();
  });
});
