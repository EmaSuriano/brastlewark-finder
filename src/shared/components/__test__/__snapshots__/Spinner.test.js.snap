// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> render 1`] = `
<i
  className=""
  color="#EF8354"
  gap={4}
  size="128px"
  theme={
    Object {
      "accent": "#EF8354",
      "background": "#4F5D75",
      "disabled": "#BFC0C0",
      "light": "#FFFFFF",
      "secondary": "#2D3142",
    }
  }
  thickness={4}
/>
`;
